### ISSUE TITLE
Server-Sent Events are broken when enabling the retry option

### ISSUE THREAD

**Issue by freud-martinou** [containous/traefik#1862]
Hello,

### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?
We expose [Dashing.io](http://dashing.io/) behind Traefik.
Dashing.io uses [Server-Sent Events](https://www.w3schools.com/html/html5_serversentevents.asp). 

If we let the default retry configuration (disabled), everything is working well.
If we enable the retry configuration (for example --retry.attempts=2), the event stream content type is replaced by Traefik by text/plain and the Browser throws an exception.

**Request example**
```
GET /events HTTP/1.1
Connection: keep-alive
Accept: text/event-stream
```

### What did you expect to see?
**Correct Response example without retry**
```
HTTP/1.1 200 OK
Content-Type: text/event-stream;charset=utf-8
Transfer-Encoding: chunked`
```
### What did you see instead?
**Incorrect Response example with --retry.attempts=2**
```
HTTP/1.1 200 OK
Content-Type: text/plain; charset=utf-8
Transfer-Encoding: chunked`
```
### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.3.3
Codename:     raclette
Go version:   go1.8.3
Built:        2017-07-06_05:32:56PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "TraefikLogsFile": "",
 "LogLevel": "ERROR",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 180000000000,
 "InsecureSkipVerify": false,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "Docker": null,
 "File": null,
 "Web": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": ""
}
```

docker-compose example
```json
{
    "version": "3",
    "services": {
        "traefik": {
            "image": "nexus.cegedim-next.net:5001/traefik:1.3.3-alpine",
            "ports": [
                "80:80",
                "8080:8080"
            ],
            "volumes": [
                "/var/run/docker.sock:/var/run/docker.sock"
            ],
            "command": "--docker --docker.swarmmode --docker.watch --web --retry.attempts=2",
            "deploy": {
                "mode": "global"
            }
        },
        "dashing": {
            "image": "johnzan/dashing",
            "deploy": {
                "labels": {
                    "traefik.enable": "true",
                    "traefik.backend": "dashing",
                    "traefik.frontend.rule": "PathPrefixStrip:/dashing",
                    "traefik.port": "3030"
                }
            }
        }
    }
}
```
Thank you!

**Comment by emilevauge** [comment]
@freud-martinou thanks for reporting. Was it working with a previous version of Traefik ?

**Comment by freud-martinou** [comment]
Hello @emilevauge,
Not to my knowledge.

We have started to use the _retry_ option with Traefik _v1.2.3_ and we encountered this problem.
So we updated to the latest Traefik release (_v1.3.3_) to see if this bug was still relevant.

**Comment by ldez** [comment]
Closed by  #2145
